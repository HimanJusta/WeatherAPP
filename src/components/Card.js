import React from 'react';
import moment from 'moment';

const Card = ({ list }) => {
  return (
    <div>
      {list.map(({ weather, dt_txt, main }) => {
        return (
          <div className="col s12 m4" key={dt_txt}>
            <div className="card blue-grey darken-1">
              <div className="card-content white-text">
                <span className="card-title left">{moment(dt_txt).format('ddd, Do MMM YYYY, hh:mm A')}</span>
                <p className="right" style={{ fontSize: '2.2rem', color: '#ffab40' }}>{`${main.temp}`} &#8451;</p>
                <img src={`http://openweathermap.org/img/w/${weather[0].icon}.png`} alt={weather[0].description}/>
                <p>{weather[0].description.toUpperCase()}</p>
              </div>
              <div className="card-action">
                <a>Max: {main.temp_max} &#8451;</a>
                <a>Min: {main.temp_min} &#8451;</a>
              </div>
            </div>
          </div>
        );
      })}
    </div>
  );
};

export default Card;
