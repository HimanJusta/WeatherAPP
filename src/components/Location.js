import React, { Component } from 'react';
import PlacesAutocomplete, {
  geocodeByAddress,
  getLatLng,
} from 'react-places-autocomplete';
import { connect } from 'react-redux';
import * as actions from '../actions';

const styles = {
  dropDownStyleActive: {
    backgroundColor: '#e6e6e6',
    cursor: 'pointer',
    boxShadow: '0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23)', padding: '10px 0 10px 10px'
  },
  dropDownStyleInActive: {
    backgroundColor: '#ffffff',
    cursor: 'pointer',
    boxShadow: '0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23)', padding: '10px 0 10px 10px'
  }
};

export class Location extends Component {
  constructor(props) {
    super(props);
    this.state = { address: '' };
  }

  handleChange = address => {
    console.log(address);
    this.setState({ address });
  };

  handleSelect = address => {
    console.log('address', address);
    this.setState({ address }, () => {
      geocodeByAddress(address)
        .then(results => getLatLng(results[0]))
        .then(latLng => {this.props.fetchWeather(latLng)})
        .catch(error => console.error('Error', error));
    });
  };

  render() {
    return (
      <PlacesAutocomplete
        value={this.state.address}
        onChange={this.handleChange}
        onSelect={this.handleSelect}
      >
        {({ getInputProps, suggestions, getSuggestionItemProps, loading }) => (
          <div>
            <input
              {...getInputProps({
                placeholder: 'Choose Location ...',
                className: 'location-search-input',
                autoFocus: true
              })}
            />
            <div className="autocomplete-dropdown-container">
              {loading && <div>Loading...</div>}
              {suggestions.map(suggestion => {
                const className = suggestion.active
                  ? 'suggestion-item--active'
                  : 'suggestion-item';
                // inline style for demonstration purpose
                const style = suggestion.active
                  ? styles.dropDownStyleActive
                  : styles.dropDownStyleInActive;
                return (
                  <div
                    {...getSuggestionItemProps(suggestion, {
                      className,
                      style,
                    })}
                  >
                    <span>{suggestion.description}</span>
                  </div>
                );
              })}
            </div>
          </div>
        )}
      </PlacesAutocomplete>
    );
  }
}

export default connect(null, actions)(Location);
