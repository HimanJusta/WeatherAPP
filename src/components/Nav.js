import React from 'react';

const Nav = () => (
  <nav>
    <div className="nav-wrapper blue darken-4">
      <a href="/" className="brand-logo center">Weather APP</a>
    </div>
  </nav>
);

export default Nav;
