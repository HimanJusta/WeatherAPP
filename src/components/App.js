import React, { Component } from 'react';
import { connect } from 'react-redux';
import Nav from './Nav';
import Location from './Location';
import Card from './Card';
import * as actions from '../actions';

export class App extends Component {
  componentDidMount() {
    this.props.fetchWeather({ lat: 12.9715987, lng: 77.59456269999998 });
  }

  render() {
    return (
      <div className="container">
        <Nav />
        <Location />
        <h2 className="center grey-text text-darken-2">{this.props.weather && this.props.weather.city ? this.props.weather.city.name : ""}</h2>
        { this.props.weather && this.props.weather.list &&
          <div className="row">
            <Card list={this.props.weather.list} />
          </div>
        }
      </div>
    );
  }
}

function mapStateToProps({ weather }) {
  return { weather };
}

export default connect(mapStateToProps, actions)(App);
