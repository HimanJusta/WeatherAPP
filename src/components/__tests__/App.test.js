import React from 'react';
import { shallow } from 'enzyme';
import { App } from '../App';
import Nav from '../Nav';
import Card from '../Card';
import Location from '../Location';

describe('App', () => {
  const props = {
    fetchWeather: jest.fn(),
    weather: {}
  };
  const app = shallow(<App {...props} />);

  it ('renders without crashing', () => {
    expect(app).toMatchSnapshot();
  });

  it('Shows a Nav', () => {
    expect(app.find(Nav).length).toEqual(1);
  });

  it('Shows a Location', () => {
    expect(app.find(Location).length).toEqual(1);
  });

  it('Does not show a Card', () => {
    expect(app.find(Card).length).toEqual(0);
  });

  it('Shows 1 Cards', () => {
    const propsWithWeather = {
      fetchWeather: jest.fn(),
      weather: {
        city: {id: 1277333, name: "Bangalore", coord: {lat: 12.9762, lon: 77.6033}, country: "IN"},
        list: [
          {
            main: {
              temp:25.02,
              temp_max:25.02,
              temp_min:23.32,
            },
            weather: [{id: 500, main: "Rain", description: "light rain", icon: "10d"}],
            dt_txt:"2018-07-11 09:00:00"
          }
        ]
      }
    };
    const appWithCard = shallow(
      <App
        {...propsWithWeather}
      />
    );
    expect(appWithCard.find(Card).length).toEqual(1);
  });
});
