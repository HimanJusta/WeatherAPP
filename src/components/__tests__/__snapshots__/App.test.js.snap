// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    fetchWeather={[Function]}
    weather={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Nav />,
        <Connect(Location) />,
        <h2
          className="center grey-text text-darken-2"
/>,
        undefined,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "center grey-text text-darken-2",
        },
        "ref": null,
        "rendered": "",
        "type": "h2",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Nav />,
          <Connect(Location) />,
          <h2
            className="center grey-text text-darken-2"
/>,
          undefined,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "center grey-text text-darken-2",
          },
          "ref": null,
          "rendered": "",
          "type": "h2",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
